src/useResizeDetector.ts

```diff
@@ -42,18 +42,38 @@ function useResizeDetector<T extends HTMLElement = any>({
     }, 0);
   }
 
-  // this is a callback that will be called every time the ref is changed
+  // this is a memo that will be called every time the ref is changed
+  // This proxy will properly call setState either when the ref is called as a function or when `.current` is set
   // we call setState inside to trigger rerender
-  const onRefChange: OnRefChangeType = useCallback(
-    (node: T | null) => {
-      if (node !== refElement) {
-        setRefElement(node);
-      }
-    },
+
+  const onRefChange: OnRefChangeType<T> = useMemo(
+    () =>
+      new Proxy(
+        node => {
+          if (node !== refElement) {
+            setRefElement(node);
+          }
+        },
+        {
+          get(target, prop) {
+            if (prop === 'current') {
+              return refElement;
+            } else {
```

Comment:
Looks like you don't need this `else`

Reply:
I added the proxy not to break anything. If I remove the else it becomes impossible to set other properties on the ref. Not sure if anyone is doing that or should be supported, but this at least keeps the behaviour the same. I could change it to not set the prop and return `false` if you think that fits better

Reply:
I meant that you don't need the "else" word itself 😅

Reply:
Ah I see 😅. Changed it